_posts/2017-06-28-Openshift-an-introduction.md

```diff
@@ -0,0 +1,241 @@
+
+__Third Party Registries:__
+OpenShift Origin can create containers using images from third party registries, but it is unlikely that these registries offer the same image notification support as the integrated OpenShift Origin registry.
+In this situation OpenShift Origin will fetch tags from the remote registry upon imagestream creation.
+
+### Master
+Managing __data storage__ is a distinct problem from managing compute resources.
+OpenShift leverages the Kubernetes PersistentVolume subsystem, which provides an __API__ for users and administrators that abstracts details of how storage is provided from how it is consumed.
+The Kubernetes pod __scheduler__ is responsible for determining placement of new pods onto nodes within the cluster.
+It reads data from the pod and tries to find a node that is a good fit based on configured policies.
+The __Management/Replication controler__ manages the lifecycle of pods.
+For instance when you deploy a new version of your application and create a new pod, Openshift can wait till the new pod is fully functional before downscaling the old pod leading to no downtime.
+__But what if the master node goes down? That's no high availability ...__ 
+You can optionally configure your masters for high availability to ensure that the cluster has no single point of failure.
+
+### Service layer
+On top of the domain and persistence layer sits the service layer of the application.
+A Kubernetes service serves as an internal load balancer.
+It identifies a set of replicated pods in order to proxy the connections it receives to them.
+Backing pods can be added to or removed from a service arbitrarily while the service remains consistently available, enabling anything that depends on the service to refer to it at a consistent internal address.
+
+### Persistant storage
+Managing storage is a distinct problem from managing compute resources. OpenShift Origin leverages the __Kubernetes persistent volume (PV)__ framework to allow administrators to provision persistent storage for a cluster.
+Using __persistent volume claims (PVCs)__, developers can request PV resources without having specific knowledge of the underlying storage infrastructure.
+PVCs are specific to a project and are created and used by developers as a means to use a PV.
+PV resources on their own are not scoped to any single project; they can be shared across the entire OpenShift Origin cluster and claimed from any project.
+After a PV has been bound to a PVC, however, that PV cannot then be bound to additional PVCs. This has the effect of scoping a bound PV to a single namespace (that of the binding project).
+
+# Openshift.io
+So before I end this blog i have to quicly mention [Openshift.io](https://openshift.io/)
+As of this moment it's not yet available but you can try to register for the preview.
+I didn't had the change yet to play with it yet since i have not yet recieved my access.
```

Comment:
I **haven't** had the chance **to** play with it, since **I** **haven't** received my access **just yet.**